### PR TITLE
Introduce installation from GitHub releases

### PR DESCRIPTION
**Pull request by nre-ableton**
This PR introduces the ability to install ccache from a statically-compiled binary via the GitHub releases page.

Note that it also contains a breaking change, namely the removal of the `ccache_build_from_sources` variable. This should instead be replaced by `ccache_install_type: "build"`.

### CODE REVIEW

**Comment by nre-ableton** [comment]
I suspect @ala-ableton will be interested in reviewing this.

File: README.md [hunk 1]
```diff
@@ -21,6 +21,9 @@ The following variables influence how `ccache` is installed on the host:
   ["Configuration options" section][ccache-config-options].
 - `ccache_install_type`: Where/how to install ccache. Valid options include:
   - `build`: Build from sources
+  - `github`: Use a precompiled binary from the ccache GitHub releases page (note: not all
+    ccache versions support all platforms. Linux binaries started appearing at release
+    4.6.1, and Mac binaries at release 4.8.)
```
Comment (rco-ableton) [inline README.md:26]: Is there a patch version missing here for the Mac binaries (`4.8.?`)?

File: defaults/main.yml [hunk 2]
```diff
@@ -11,11 +11,14 @@ ccache_config_options: {}
...
 
-# Where to install ccache when building from sources
+# Where to install ccache when building from sources or installing precompiled binaries
 ccache_install_prefix: "/usr/local"
 
 # Where/how to install ccache. Valid options include:
 # - build: Build from sources
+# - github: Use a precompiled binary from the ccache GitHub releases page (note: not all
+#   ccache versions support all platforms. Linux binaries started appearing at release
+#   4.6.1, and Mac binaries at release 4.8.)
```
Comment (rco-ableton) [inline defaults/main.yml:21]: Same question here, is there a missing patch version?

File: molecule/focal-github/molecule.yml [hunk 3]
```diff
@@ -0,0 +1,13 @@
+---
+dependency:
+  name: galaxy
+driver:
+  name: docker
+platforms:
+  - name: instance
+    image: ubuntu:20.04
```
Comment (rco-ableton) [inline molecule/focal-github/molecule.yml]: I wondering if splitting this into an `image` and `tag` would be overkill?

File: README.md (hunk 1 above)
Comment (nre-ableton) [reply README.md:26]: No, it's not: https://github.com/ccache/ccache/releases/tag/v4.8 :shrug:

File: defaults/main.yml (hunk 2 above)
Comment (nre-ableton) [reply defaults/main.yml:21]: See above.

File: README.md (hunk 1 above)
Comment (ala-ableton) [reply README.md:26]: @nre-ableton I think what @rco-ableton finds weird is the trailing period.
```
    4.6.1, and Mac binaries at release 4.8)
```
would probably be less confusing.

File: molecule/focal-github/molecule.yml (hunk 3 above)
Comment (nre-ableton) [reply molecule/focal-github/molecule.yml]: I think that the schema for this file is controlled by the Molecule Docker driver (or whatever they refactored it into). Stupidly, that project [doesn't publish documentation anywhere](https://github.com/ansible-community/molecule-plugins/issues/20), so I can't confirm this. But basically, I don't think we can split this.

Regardless, what we _should_ do is use a shared base config, so we could avoid having to specify the image entirely (see how `ansible-consul` does it: https://github.com/ansible-community/ansible-consul/tree/master/molecule). We have an [issue open for that](https://github.com/AbletonAppDev/devtools/issues/4855), but it's in the purgatory/icebox column until we get time.

File: README.md (hunk 1 above)
Comment (rco-ableton) [reply README.md:26]: Yes, it was the trailing dot—I assumed it was a version separator.

File: molecule/focal-github/molecule.yml (hunk 3 above)
Comment (rco-ableton) [reply molecule/focal-github/molecule.yml]: Makes sense. I missed that this was part of `molecule`. Sorry for the noise!

File: README.md (hunk 1 above)
Comment (nre-ableton) [reply README.md]: Fixed in 6da0ef9.

File: defaults/main.yml (hunk 2 above)
Comment (nre-ableton) [reply defaults/main.yml]: Fixed in 6da0ef9.

File: README.md (hunk 1 above)
Comment (nre-ableton) [reply README.md]: Ping @rco-ableton (sorry, I can't re-request a review).

Comment (rco-ableton) [reply README.md]: Thanks!

**Comment by nre-ableton** [comment]
@ala-ableton Your turn!

File: defaults/main.yml [hunk 4]
```diff
@@ -18,6 +14,11 @@ ccache_group: "{{ ansible_user }}"
 # Where to install ccache when building from sources
 ccache_install_prefix: "/usr/local"
 
+# Where/how to install ccache. Valid options include:
```
[the remark this answers is not in the document]
Comment (nre-ableton) [reply defaults/main.yml]: See d774097. Also a79a243 for the documentation of `ccache_install_prefix`.

**Comment by nre-ableton** [comment]
@ala-ableton Let me know when I should rebase.

**Review by ala-ableton — APPROVED** [review]

**Pull request merged**